h/services/jwt.py

```diff
@@ -47,7 +56,152 @@ def decode_token(
             )
         except PyJWTError as err:
             msg = f"Invalid JWT for: {iss}, {aud}. {err}"
-            raise TokenValidationError(msg) from err
+            raise JWTDecodeError(msg) from err
+
+    def encode_symmetric(
+        self,
+        payload,
```

Comment:
`payload` can be an instance of any dataclass (but it must be a dataclass: we call `dataclass.asdict()` on it below and this only works for dataclasses). I don't think there's any way to type-annotation "instance of any dataclass", leaving it to the docstring to explain.

Reply:
Indeed I can't see a protocol that represents a dataclass in the Python standard library. There is one that mypy has which it uses to type functions like `asdict`:

```py
from dataclasses import dataclass
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from _typeshed import DataclassInstance


@dataclass
class IAmADataclass:
    foo: int


class NotADataclass:
    foo: int


def wants_a_dataclass(x: DataclassInstance):
    pass


wants_a_dataclass(IAmADataclass(foo=1))  # OK
wants_a_dataclass(NotADataclass())  # Typecheck error
```

A docstring here is fine though.